src/sentry/integrations/opsgenie/integration.py

```diff
@@ -51,8 +62,92 @@
 )
 
 
+class OpsgenieFinishedView(PipelineView):
+    def dispatch(self, request: Request, pipeline: Pipeline) -> Response:
+        response = pipeline.finish_pipeline()
+
+        integration = getattr(pipeline, "integration", None)
+        if not integration:
+            return response
+
+        messages.add_message(request, messages.SUCCESS, "Opsgenie integration installed.")
+
+        return self.redirect(
+            absolute_uri(
+                f"/settings/{pipeline.organization.slug}/integrations/opsgenie/{integration.id}/"
+            )
+        )
+
+
+class InstallationForm(forms.Form):
+    base_url = forms.CharField(
+        label=_("Base URL"),
+        help_text=_("Either https://api.opsgenie.com/ or https://api.eu.opsgenie.com/"),
+        widget=forms.TextInput(attrs={"placeholder": "https://api.opsgenie.com/"}),
+    )
+    api_key = forms.CharField(
+        label=("Opsgenie API Key"),
+        widget=forms.TextInput(
+            attrs={
+                "placeholder": _("5832fc6e14300a0d962240a8144466eef4ee93ef0d218477e55f11cf12fc3737")
+            }
+        ),
+    )
+
+
+class InstallationConfigView(PipelineView):
+    def dispatch(self, request: Request, pipeline) -> HttpResponse:
```

Comment:
Some small typing fixes that mypy complains about b/c the parent class returns Http instead of base request stuff. You can also run `mypy` in your console to check if you've caught all of the typing issues. We just had a big python typing change so sometimes random typing things break and you have to root out the issue :sweat:
```suggestion
    def dispatch(self, request: HttpRequest, pipeline) -> HttpResponse:
```

Reply:
Running into issues where the PipelineView has `def dispatch(self, request: Request, pipeline) -> HttpResponse:` but the BaseView has `def dispatch(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponseBase`. I've added a `# type:ignore` for now to silence the errors.